From e31b1e8604c22102fb95560a6b1656c8ae9070f6 Mon Sep 17 00:00:00 2001
From: Paul Harrison <pharrison@mythtv.org>
Date: Tue, 25 Mar 2014 16:12:46 +0000
Subject: [PATCH 6/6] MythMusic: add a check for the AlbumArt storage group

Don't show any of the music screens before a valid AlbumArt directory is
defined. We already check for the Music storage group so this just extends
that.
---
 mythplugins/mythmusic/mythmusic/main.cpp |   23 +++++++++++++++++++++++
 1 file changed, 23 insertions(+)

diff --git a/mythplugins/mythmusic/mythmusic/main.cpp b/mythplugins/mythmusic/mythmusic/main.cpp
index 5b632e7..41a0540 100644
--- a/mythplugins/mythmusic/mythmusic/main.cpp
+++ b/mythplugins/mythmusic/mythmusic/main.cpp
@@ -93,6 +93,29 @@ static bool checkStorageGroup(void)
        return false;
     }
 
+    // get a list of hosts with a directory defined for the 'MusicArt' storage group
+    hostList.clear();
+    sql = "SELECT DISTINCT hostname "
+                  "FROM storagegroup "
+                  "WHERE groupname = 'MusicArt'";
+    if (!query.exec(sql) || !query.isActive())
+        MythDB::DBError("checkStorageGroup get host list", query);
+    else
+    {
+        while(query.next())
+        {
+            hostList.append(query.value(0).toString());
+        }
+    }
+
+    if (hostList.isEmpty())
+    {
+        ShowOkPopup(qApp->translate("(MythMusicMain)",
+                                    "No directories found in the 'MusicArt' storage group. "
+                                    "Please run mythtv-setup on the backend machine to add one."));
+       return false;
+    }
+
     return true;
 }
 
-- 
1.7.10.2

